src/main/java/com/redhat/jenkins/plugins/bayesian/Utils.java

```diff
@@ -45,10 +44,23 @@
         // Maven
         List<FilePath> mavenManifests = findManifestsFromList(workspace, knownMavenManifests);
         if (!mavenManifests.isEmpty()) {
-            // TODO: get all poms
-            FilePath pom = workspace.child("target/stackinfo/poms/pom.xml");
-            if (manifestExists(pom)) {
-                manifests.add(pom);
+            List<FilePath> childPomDirs = new ArrayList<FilePath>();
+            FilePath rootPom = workspace.child("target/stackinfo/poms/pom.xml");
+            if (manifestExists(rootPom)) {
+                manifests.add(rootPom);
+            }
+
+            FilePath childPomDirsPath = workspace.child("target/stackinfo/poms");
+            try {
+                childPomDirs = childPomDirsPath.listDirectories();
```

Comment:
Yes msrb. it covers the first level only. I will update it to cover the directories recursively. Thanks for catching this.